ISISMuon/MuonDataLib
# Loads and basic plots for partial event data

### Description of work
This work adds the code needed to load in event data as produced by @modularius and some basic plotting.

I have added tests for everything except for the plotting. It is unclear if this will just be a stop gap solution. However, it has been added to make it easier for users. 

Part Fixes #10. 


### To test:


It does require a conda environment update as I have added plotly (or you could pip install). 

Running the following code should give some nice plots in a browser (clicking the label in the legend will show/hide the data). 

```
from MuonDataLib.plot.basic import Figure
from MuonDataLib.data.loader.load_event import LoadEventData
import os

# set up data to plot
file = os.path.join(os.path.dirname(__file__),
                    'test',
                    'data_files',
                    'complete_run.nxs')


file_0 = os.path.join(os.path.dirname(__file__),
                    'test',
                    'data_files',
                    'partial_run.nxs')

# create data loaders and get histograms
data_p = LoadEventData()
data_p.load_data(file_0)

data_f = LoadEventData()
data_f.load_data(file)

# create plots
fig = Figure('compare part and full')
# these plot the 0, 1 and 2 detector ID's
fig.plot_from_instrument(data_p, [0,1,2], 'part')
fig.plot_from_instrument(data_f, [0,1,2], 'full')
fig.show()
```

To test the reload:
```
from MuonDataLib.plot.basic import Figure
from MuonDataLib.data.loader.load_event import LoadEventData
import os

# set up data to plot
file = os.path.join(os.path.dirname(__file__),
                    'test',
                    'data_files',
                    'complete_run.nxs')


file_0 = os.path.join(os.path.dirname(__file__),
                    'test',
                    'data_files',
                    'partial_run.nxs')

# create data loaders and get histograms
data_p = LoadEventData()
data_p.load_data(file_0)

# mock reload by manually changing file
data_p._file_name = file
data_p.reload_data()

# create plots
fig = Figure('compare part and full')
# these plot the 0, 1 and 2 detector ID's
fig.plot_from_instrument(data_p, [0,1,2], 'Reload')

fig.show()
```


#### Code Review

- Is the code of an acceptable quality?
- Are the unit tests small and test the class in isolation?
- Does the documentation build [ReadTheDocs](https://readthedocs.org/projects/muondatalib/builds/)?

#### Functional Tests

- Do changes function as described? Add comments below that describe the tests performed?
- Do the changes handle unexpected situations, e.g. bad input?
- Has the relevant (user and developer) documentation been added/updated?

Does everything look good? Mark the review as **Approve**.
